### ISSUE TITLE
SSH Failed in recent build - Process tried to write to a nonexistent pipe.

### ISSUE THREAD

**Issue by scmcduffee** [microsoft/vscode-remote-release#1887]
- VSCode Version: 1.40.1
- Local OS Version: Win 10
- Remote OS Version: Ubuntu 18.04
- Remote Extension/Connection Type: SSH

ssh connection has recently started failing.  It was working fine before recent updates.  Connection to the remote machine is still possible outside of VSCode.

Output:

```
[15:10:59.423] remote-ssh@0.47.2
[15:10:59.423] win32 x64
[15:10:59.424] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a2273636d2d77732d6f6e496e74656c4e6574227d", attempt 1
[15:10:59.424] SSH Resolver called for host: scm-ws-onIntelNet
[15:10:59.425] Setting up SSH remote "scm-ws-onIntelNet"
[15:10:59.469] Using commit id "8795a9889db74563ddd43eb0a897a2384129a619" and quality "stable" for server
[15:10:59.470] Testing ssh with ssh -V
[15:10:59.647] ssh exited with code: 0
[15:10:59.648] Got stderr from ssh: OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[15:10:59.652] Running script with connection command: ssh -T -D 54200 scm-ws-onIntelNet bash
[15:10:59.654] Install and start server if needed
[15:10:59.657] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[15:10:59.962] > 
[15:10:59.962] Got some output, clearing connection timeout
[15:11:00.131] > CreateProcessW failed error:87
> posix_spawn: Unknown error
> The process tried to write to a nonexistent pipe.
> 
[15:11:00.446] "install" terminal command done
[15:11:00.446] Install terminal quit with output: The process tried to write to a nonexistent pipe.
[15:11:00.446] Received install output: The process tried to write to a nonexistent pipe.
[15:11:00.446] Resolver error: The process tried to write to a nonexistent pipe
[15:11:00.449] TELEMETRY: {"eventName":"resolver","properties":{"outcome":"failure","reason":"OfflineError","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","gotUnrecognizedPrompt":"0","remoteInConfigFile":"1"},"measures":{"resolveAttempts":1,"retries":1}}
[15:
```

**Comment by roblourens** [comment]
So does it work to run a command like `echo uname | ssh -T -D 54200 scm-ws-onIntelNet bash`? Does -T make a difference?

**Comment by scmcduffee** [comment]
Hi Thanks for getting back to me.  Yes that works but VSCode connection does not.  It used to but the current version broke.

**Comment by roblourens** [comment]
Nothing has changed on our end. Can you try searching for that error and see whether you see anything relevant? e.g. https://serverfault.com/questions/956613/windows-10-ssh-proxycommand-posix-spawn-no-such-file-or-directory

The real error is `posix_spawn: Unknown error`

**Comment by scmcduffee** [comment]
Hmm I do use the full path name - however I am using mingw64's connect.exe for the ProxyCommand

**Comment by scmcduffee** [comment]
Actually I'm sorry - the ssh command you gave me to try does not work.  I missed it somehow.  Errors out with: Warning: untrusted X11 forwarding setup failed: xauth key data not generated
Linux.  Something with the DISPLAY env var?  Hmm - I wonder what changed to start it failing.

**Comment by scmcduffee** [comment]
Commenting out my ForwardX11 yes line still does not allow me to connect through VSCode however the command line option does work.

**Comment by scmcduffee** [comment]
So I fixed this.  It indeed was not a problem with VSCode but a problem with the OpenSSH installed from Windows 10 through the Apps & Features > Manage Optional Settings > OpenSSH Client.  I uninstalled that client and instead installed the OpenSSH managed from the PowerShell git page: https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH.  Downloaded the latest from there and added the location to my PATH and I'm all good.
